src/chat_do.py

```
@@ -0,0 +1,337 @@
+        return _decrypt_xor(ciphertext, secret)
+    raw = base64.b64decode(ciphertext[3:])
+    iv, ct = raw[:12], raw[12:]
+    key_bytes = _pbkdf2_key(secret)
+    crypto_key = await _import_aes_key(key_bytes)
+    iv_array = to_js(iv, create_pyproxies=False)
+    algo = to_js({"name": "AES-GCM", "iv": iv_array}, dict_converter=js.Object.fromEntries)
+    data = to_js(ct, create_pyproxies=False)
+    try:
+        pt_buf = await js.crypto.subtle.decrypt(algo, crypto_key, data)
+        return bytes(js.Uint8Array.new(pt_buf)).decode("utf-8")
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)